scipy/stats/_stats_py.py

```diff
@@ -70,10 +70,16 @@
 from scipy import stats
 from scipy.optimize import root_scalar
 from scipy._lib._util import normalize_axis_index
-from scipy._lib._array_api import (array_namespace, is_numpy, atleast_nd,
-                                   xp_clip, xp_moveaxis_to_end, xp_sign,
-                                   xp_minimum, xp_vector_norm)
-from scipy._lib.array_api_compat import size as xp_size
+from scipy._lib._array_api import (
+    _asarray,
```

Comment:
Why change formatting and not import as `xp_array` anymore?

Reply:
Where was the `xp_array`? I've just added the import of `_asarray`, and imported `size` from `_array_api` rather than `array_api_compat`.

The formatting change is just to improve readability. IMO this is a useful thing to do when the lines are already being touched (while we wait for auto-formatting to materialise), but I can see the argument for keeping the diff as minimal as possible. In this case, it only takes a few seconds to check.

Reply:
My concern is that argument can be made for any change that formats a la Black, which is very polarizing. My opinions about it are mixed - if we adopted it as a project, I'd understand and be OK with it. But we haven't yet, so I'd prefer not to set a precedent that unnecessary Black format changes are ok. You're a maintainer, too, so you can do it if you want, but I might still say something about it.

I thought `_asarray` was already imported as `xp_asarray`, but I guess not. I do try to do that when I import things. You know, for "readability" : )

Reply:
I'm happy to review or submit a PR to change `_asarray` to `xp_asarray`! I think I chose `_asarray` to be as close to `asarray` as possible, but actually I do agree that `xp_asarray` is more readable.

> I might still say something about it.

Me too :) until we have a project-wide decision I think the sensible thing is to accept either way. I think there is a substantial difference between somebody sending PRs specifically for opinionated style cleanups which we haven't agreed upon (indeed, we would not want to set a precedent for that), and a contributor making the imports of the files they are working on easier for them to read (when the lines are already diffed).

Reply:
Agreed with the `black` concern - nicer to not change styles.

> I'm happy to review or submit a PR to change `_asarray` to `xp_asarray`! I think I chose `_asarray` to be as close to `asarray` as possible, but actually I do agree that `xp_asarray` is more readable.

I will note that we discussed this name twice already now, first when naming it `as_xparray`, then with the rename to `_asarray`. See for example https://github.com/scipy/scipy/issues/18866#issuecomment-1634998051.

In general, our `xp_func` names are matching behavior with the `xp.func` equivalent, and we're putting them into `_lib` when things aren't yet available upstream etc. This function on the other hand has a lot of extra keywords and semantics, and does not match `xp.asarray`. 

So that's two reasons for leaving it as is - that'd be better imho.

Reply:
Re the reformatting: I think the point made about not overdoing `black`-style reformatting is valid and good to keep in mind. But now that the change is made here and the code certainly doesn't look worse (arguably more readable indeed), keeping this change seems fine to me.